pkg/repository/maintenance.go

```diff
@@ -262,3 +294,97 @@ func getLatestMaintenanceJob(cli client.Client, ns string) (*batchv1.Job, error)
 
 	return &jobList.Items[0], nil
 }
+
+// getMaintenanceJobConfig is called to get the Maintenance Job Config for the
+// BackupRepository specified by the repo parameter.
+//
+// Params:
+//
+//	veleroNamespace: the Velero-installed namespace. It's used to retrieve the BackupRepository.
+//	repo: the BackupRepository to get the Job config.
+func getMaintenanceJobConfig(
+	ctx context.Context,
+	cli client.Client,
+	logger logrus.FieldLogger,
+	veleroNamespace string,
+	repoMaintenanceJobConfig string,
+	repo *velerov1api.BackupRepository,
+) (*JobConfigs, error) {
+	var cm v1.ConfigMap
+	if err := cli.Get(
+		ctx,
+		types.NamespacedName{
+			Namespace: veleroNamespace,
+			Name:      repoMaintenanceJobConfig,
+		},
+		&cm,
+	); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		} else {
+			return nil, errors.Wrapf(
+				err,
+				"fail to get repo maintenance job configs %s", repoMaintenanceJobConfig)
+		}
+	}
+
+	if cm.Data == nil {
+		return nil, errors.Errorf("data is not available in config map %s", repoMaintenanceJobConfig)
+	}
+
+	configMap := make(map[string]JobConfigs)
+	for k, v := range cm.Data {
+		var config JobConfigs
+		if err := json.Unmarshal([]byte(v), &config); err != nil {
+			return nil, errors.Wrapf(err, "fail to unmarshal configs from %s", repoMaintenanceJobConfig)
+		}
+		configMap[k] = config
+	}
+
+	// Generate the BackupRepository key.
+	// If using the BackupRepository name as the is more intuitive,
+	// but the BackupRepository generation is dynamic. We cannot assume
+	// they are ready when installing Velero.
+	// Instead we use the volume source namespace, BSL name, and the uploader
+	// type to represent the BackupRepository. The combination of those three
+	// keys can identify a unique BackupRepository.
+	repoJobConfigKey := repo.Spec.VolumeNamespace + "-" +
+		repo.Spec.BackupStorageLocation + "-" + repo.Spec.RepositoryType
+
+	var result *JobConfigs
+	if _, ok := configMap[repoJobConfigKey]; ok {
+		logger.Debugf("Find the repo maintenance config %s for repo %s", repoJobConfigKey, repo.Name)
+		result = new(JobConfigs)
+		*result = configMap[repoJobConfigKey]
+	}
+
+	if _, ok := configMap["global"]; ok {
+		logger.Debugf("Find the global repo maintenance config for repo %s", repo.Name)
+
+		if result == nil {
+			result = new(JobConfigs)
+		}
+
+		if result.CPULimit == "" {
```

Comment:
When the values in PodResource are all empty (strings == ""), it also means a validate configuration --- BestEffort, so I don't look a correct manner to use the global configuration when the private one is empty.
It may be better to define the PodResource as a pointer.

Additionally, we normally regard the PodResource as a whole, at least we regard the memory request/limit or the CPU request/limit as a whole, so it is confusing to get part of it from the private value and the other part from the global value.